effekt/shared/src/main/scala/effekt/core/vm/Builtin.scala

```diff
@@ -267,4 +333,17 @@ protected object As {
       case _ => None
     }
   }
+  object Regex {
+    def unapply(v: Value): Option[regex.Regex] = v match {
+      case Value.Literal(v: regex.Regex) => Some(v)
+      case _ => None
+    }
+  }
+  object RegexMatch {
+    def unapply(v: Value): Option[regex.Regex.Match | Null] = v match {
+      case Value.Literal(null) => Some(null)
```

Comment:
I am not entirely happy about using `Literal` for arbitrary extern values, but well...